File: Problems/MeetingScheduler.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution
{
public:
    vector<int> minAvailableDuration(vector<vector<int>> &slots1, vector<vector<int>> &slots2, int duration)
    {
        sort(slots1.begin(), slots1.end());
        sort(slots2.begin(), slots2.end());
        int count1 = 0, count2 = 0;
        while (slots1.size() > count1 && slots2.size() > count2)
        {
            vector<int> slot1 = slots1[count1], slot2 = slots2[count2];
            int dur = min(slot1[1], slot2[1]) - max(slot1[0], slot2[0]);
            int start = max(slot1[0], slot2[0]);
            if (dur >= duration)
                return {start, start + duration};
            if (slot2[1] > slot1[1])
                count1++;
            else
                count2++;
        }
        return vector<int>();
    }
};

File: Problems/MeetingScheduler.py
from typing import List


class Solution:
    def minAvailableDuration(self, slots1: List[List[int]], slots2: List[List[int]], duration: int) -> List[int]:
        slots1.sort()
        slots2.sort()
        count1, count2 = 0, 0
        while len(slots1) > count1 and len(slots2) > count2:
            slot1 = slots1[count1]
            slot2 = slots2[count2]
            dur = min(slot1[1], slot2[1]) - max(slot1[0], slot2[0])
            start = max(slot1[0], slot2[0])
            if dur >= duration:
                return [start, start + duration]
            if slot1[1] > slot2[1]:
                count2 += 1
            elif slot1[1] < slot2[1]:
                count1 += 1
            else:
                count1 += 1
                count2 += 1
        return []
